### ISSUE TITLE
Parsing not compliant with the spec

### ISSUE THREAD

**Issue by EECOLOR** [postcss/postcss-selector-parser#121]
[A line from the tests](https://github.com/postcss/postcss-selector-parser/blob/master/src/__tests__/attributes.js#L10) that shows the of result parsing an attribute:

```js
t.deepEqual(tree.nodes[0].nodes[0].attribute, '  href');
```

[Another line from the tests](https://github.com/postcss/postcss-selector-parser/blob/master/src/__tests__/attributes.js#L244) that shows the of result parsing a value:

```js
t.deepEqual(tree.nodes[0].nodes[1].value, '"test"  ');
```

In the above examples the spaces are included in the parsed attribute and value. The [spec (or CSS grammar)](https://www.w3.org/TR/CSS21/grammar.html) however states the following:

```
attrib
  : '[' S* IDENT S* [ [ '=' | INCLUDES | DASHMATCH ] S*
    [ IDENT | STRING ] S* ]? ']'
```
This means that spaces should not be part of the attribute, operator or value.

**Comment by EECOLOR** [comment]
I forgot to mention that the parser for operator was changed in response to this issue: https://github.com/postcss/postcss-selector-parser/issues/118

**Comment by chriseppstein** [comment]
Agree. this is wrong.

**Comment by chriseppstein** [comment]
Fixed by 872559f545f75f9220e627ec53d7602082eed5ae

**Comment by EECOLOR** [comment]
@chriseppstein Wow, that is some serious piece of work. Thanks!
